enterprise/utils.py

```diff
@@ -1381,6 +1389,127 @@ def enroll_user(enterprise_customer, user, course_mode, *course_ids, **kwargs):
+    license_uuid=None,
+):
+    """
+    Get or Create the Enterprise Course Enrollment.
+
+    If ``license_uuid`` present, will also create a LicensedEnterpriseCourseEnrollment record.
+    """
+    source = enterprise_enrollment_source_model().get_source(enterprise_enrollment_source_model().ENROLLMENT_URL)
+    # Create the Enterprise backend database records for this course
+    # enrollment
+    enterprise_course_enrollment, created = enterprise_course_enrollment_model.objects.get_or_create(
+        enterprise_customer_user=enterprise_customer_user,
+        course_id=course_id,
+        defaults={
+            'source': source
+        }
+    )
+    if license_uuid and not enterprise_course_enrollment.license:
+        LOGGER.info(
+            'LicensedEnterpriseCourseEnrollment being created with following info: '
+            'License UUID: {license_uuid}, '
+            'EnterpriseCourseEnrollment: {enterprise_course_enrollment_uuid}'.format(
+                license_uuid=license_uuid,
+                enterprise_course_enrollment_uuid=enterprise_course_enrollment,
+            )
+        )
+        licensed_enterprise_course_enrollment_model().objects.create(
+            license_uuid=license_uuid,
+            enterprise_course_enrollment=enterprise_course_enrollment,
+        )
+    return enterprise_course_enrollment, created
+
+
+def enroll_licensed_users_in_courses(enterprise_customer, licensed_users_info, discount=100.00):
+    """
+    Takes a list of licensed learner data and enrolls each learner in the requested courses.
+
+    Args:
+        - enterprise_customer: The EnterpriseCustomer (object) which is sponsoring the enrollment
+        - licensed_users_info: (list) An array of dictionaries, each containing information necessary to create a
+            licensed enterprise enrollment for a specific learner in a specified course run.
+            Example:
+                licensed_users_info: [
+                    {
+                        'email': 'newuser@test.com',
+                        'course_run_key': 'course-v1:edX+DemoX+Demo_Course',
+                        'course_mode': 'verified',
+                        'license_uuid': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    }
+                ]
+        - discount: (int) the discount offered to the learner for their enrollment. Subscription based enrollments
+            default to 100
+
+    Expected Return Values:
+        Results: {
+            successes: [{ 'email': <email>, 'course_run_key': <key>, 'user': <user object> } ... ],
+            pending: [{ 'email': <email>, 'course_run_key': <key>, 'user': <user object> } ... ],
+            failures: [{ 'email': <email>, 'course_run_key': <key> } ... ]
+        }
+    """
+    results = {
+        'successes': [],
+        'pending': [],
+        'failures': [],
+    }
+    for licensed_user_info in licensed_users_info:
+        user = User.objects.filter(email=licensed_user_info['email']).first()
```

Comment:
isn't there a more specific, find a user by email query instead? not clear if expect multiple User object per email....

Reply:
I could probably use a `.get()` here, right? But also I don't think you can have multiple accounts to one email?

Reply:
yeah figured there was no way, since email should be a unique key for User I hope! but this works too, just want to check if there is a more optimal way here, sounds like get could work (if user is invalid it might just throw right away)